exporters/committime/collector_base.py

```diff
@@ -178,11 +198,25 @@ def get_metric_from_build(self, build, app, namespace, repo_url):
             if not repo_url:
                 if build.spec.source.git:
                     repo_url = build.spec.source.git.uri
+                elif build.metadata.labels.pelorusgithost:
+                    git_host = build.metadata.labels.pelorusgithost
+                    git_project = build.metadata.labels.pelorusgitproject
+                    git_app = build.metadata.labels.pelorusgitapp
+                    git_protocol = build.metadata.labels.pelorusgitprotocol
+                    repo_url = (
+                        git_protocol
+                        + "://"
+                        + git_host
+                        + "/"
+                        + git_project
+                        + "/"
+                        + git_app
+                    )
```

Comment:
I don't think we should break up git URLs in this way. There are a myriad of [supported git url formats](https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats) that are not covered by this structure, and to try to cover them all would be cumbersome and unnecessary. I would just stick to the two standard git parameters that are used all over the place:

- `git_url` - The full URL string in any format
- `git_ref` - The branch or tag to be examined, if applicable

Reply:
@etsauer I initially approached the solution using labels. But labels have character restrictions for the value part. The idea is to improve this with an annotation to avoid dirty code like this one.

Reply:
Using annotations makes sense. I'll add a `BUILD_GIT_REPO_URI_ANNOTATION` env variable (or is that too long...)

We should probably use a namespaced default, right? Something like `pelorus.konveyor.io/git-uri`? Or keep it simple with `gitUri`?

Reply:
@KevinMGranger I was updating these values with a pipeline dynamically. Regarding the env variable, I think it is fine, I prefer it to be long and understandable rather than short and confusing.

Regarding the namespaced default, I think it's the same thing... keeping it simple can lead to confusion, I think having pelorus word in the name would indicate what it is.